docs/tutorials/semantic_search/generate_embeddings_for_arrays_of_object.md

```diff
@@ -0,0 +1,299 @@
+# Topic
+
+This tutorial shows how to generate embeddings for arrays of objects in OpenSearch.
+
+Note: Replace the placeholders that start with `your_` with your own values.
+
+# Steps
+
+## 1. Create embedding model
+
+We will use [Bedrock Titan Embedding model](https://docs.aws.amazon.com/bedrock/latest/userguide/titan-embedding-models.html) in this tutorial. 
+
+- If you are using AWS managed OpenSearch service, you can use this [python notebook](https://github.com/opensearch-project/ml-commons/blob/2.x/docs/tutorials/aws/AIConnectorHelper.ipynb) to create Bedrock Embedding Model easily. Search `1. Create Connector of Bedrock Embedding Model` on the page. 
+Or you can manually create connector following this [tutorial](https://github.com/opensearch-project/ml-commons/blob/2.x/docs/tutorials/aws/semantic_search_with_bedrock_titan_embedding_model.md).
+
+- If you are using self-managed OpenSearch, you can follow this [blueprint](https://github.com/opensearch-project/ml-commons/blob/2.x/docs/remote_inference_blueprints/bedrock_connector_titan_embedding_blueprint.md). 
+
+Use the model ID from the response to test predict API:
+```
+POST /_plugins/_ml/models/your_embedding_model_id/_predict
+{
+    "parameters": {
+        "inputText": "hello world"
+    }
+}
+```
+Sample response:
+
+```
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "sentence_embedding",
+          "data_type": "FLOAT32",
+          "shape": [ 1536 ],
+          "data": [0.7265625, -0.0703125, 0.34765625, ...]
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+
+## 2. Create ingest pipeline
+
+### 2.1 Create test index
+```
+PUT my_books
+{
+  "settings" : {
+      "index.knn" : "true",
+      "default_pipeline": "bedrock_embedding_foreach_pipeline"
+  },
+  "mappings": {
+    "properties": {
+      "books": {
+        "type": "nested",
+        "properties": {
+          "title_embedding": {
+            "type": "knn_vector",
+            "dimension": 1536
+          },
+          "title": {
+            "type": "text"
+          },
+          "description": {
+            "type": "text"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### 2.2 Create ingest pipeline
+
+Create sub-pipeline to generate embedding for one item in the array.
+
+This pipeline contains 3 processors
+- set processor: The `text_embedding` processor is unable to identify "_ingest._value.title". You need to copy "_ingest._value.title" to a temporary field for text_embedding to process it.
```

Comment:
how customers will know what will be the name of the temporary field? if the original field name is `xyz` then embedding field will be `xyz_embedding`?

Reply:
It could be arbitrary name, just make sure it's not same with any field which is being used in document.